actioncable/lib/action_cable/engine.rb

```diff
@@ -22,7 +22,9 @@ class Railtie < Rails::Engine # :nodoc:
 
     initializer "action_cable.set_configs" do |app|
       options = app.config.action_cable
-      options.allowed_request_origins ||= "http://localhost:3000" if ::Rails.env.development?
+      if ::Rails.env.development? && !ActionCable.server.config.allowed_request_origins
+        options.allowed_request_origins ||= "http://localhost:3000"
+      end
```

Comment:
Put another way, we shouldn't access Action Cable before it's `on_load` hook is fired. That's what it's there for 😁